### ISSUE TITLE
Feature request : family   named parameter:

### ISSUE THREAD

**Issue by samithe7** [rrousselGit/riverpod#2123]
Now   family  is not a    named parameter 
 

```dart

 final sss = StateProvider.autoDispose.family<String? ,String>((ref , name) {
    return null;
  });

// 

class ShowWidget extends ConsumerWidget {
  const ShowWidget({
    super.key,

  });





  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final sami = ref.watch(sss( name:  secName));
    return Text(sami);
  }
}


```

**Comment by AhmedLSayed9** [comment]
This is possible and one of the benefits of code gen: https://docs-v2.riverpod.dev/docs/about_code_generation

**Comment by samithe7** [comment]
is it 

> This is possible and one of the benefits of code gen: https://docs-v2.riverpod.dev/docs/about_code_generation

thanks . is it  possible  without  code gen?

**Comment by AhmedLSayed9** [comment]
> thanks . is it possible without code gen?

No. Without code gen, you can use only one positional parameter, with code gen, you can use multiple named, optional or default parameters.

**Comment by rrousselGit** [comment]
This isn't possible without codegen. It requires parsing the function to be able to know what the provider requests.

But codegen already supports this, so I'll close this.
